snare/cloner.py

```diff
@@ -144,7 +153,64 @@ def _make_filename(self, url):
         hash_name = m.hexdigest()
         return file_name, hash_name
 
-    async def get_body(self, session):
+    async def get_site(self, session, current_url):
+        """
+        Fetch the given URL with aiohttp
+        """
+        data = None
+        headers = []
+        content_type = None
+        try:
+            response = await session.get(
+                current_url,
+                headers={
+                    "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:89.0) Gecko/20100101 Firefox/89.0",
+                    "Accept": "text/html",
+                },
+                timeout=10.0,
+            )
+            headers = self.get_headers(response.headers)
+            content_type = response.content_type
+            data = await response.read()
+            data = data.decode()
+        except (aiohttp.ClientError, asyncio.TimeoutError) as client_error:
+            self.logger.error(client_error)
+        else:
+            await response.release()
+        return [data, headers, content_type]
+
+    async def get_site_headless(self, driver, session, current_url):
+        """
+        Fetch the given URL with selenium (headless cloning)
+        """
+        data = None
+        headers = []
+        content_type = None
+        try:
+            driver.get(str(current_url))
+        except TimeoutException:
+            self.logger.error("Request timed out:" + str(current_url))
+        except InvalidArgumentException:
+            self.logger.error("Malformed URL:" + str(current_url))
+        except Exception as err:
+            self.logger.error(err)
+        time.sleep(1)
+        data = driver.page_source
```

Comment:
do we want to stick to async version? All our functions are async, and I think it makes sense to use async libs like https://github.com/pyppeteer/pyppeteer
@lordlabuckdas @mzfr what do you think?